tls/ttls.c

```diff
@@ -46,6 +46,12 @@ static DEFINE_PER_CPU(struct aead_request *, g_req) ____cacheline_aligned;
 static struct kmem_cache *ttls_hs_cache = NULL;
 static ttls_send_cb_t *ttls_send_cb;
 
+static inline size_t
+ttls_max_ciphertext_len(const TlsXfrm *xfrm)
+{
+	return TLS_MAX_PAYLOAD_SIZE + xfrm->minlen;
```

Comment:
This is crucial limit - please reference appropriate RFC in a comment. `minlen` is typically very small as being initialized in `ttls_derive_keys()` and it seems never reaches 2048 as described in the PR comment.

Reply:
I've added a comment with description to the patch.

In short, there are limits on plaintext size (2^14 bytes) and on ciphertext size (2^14 + 2048). But since we use ciphers that add a constant overhead to each record, there is no way 2^14 + 2048 bytes ciphertext after deciphering will fit into 2^14 byte limit of a plaintext. So the limit is 2^14 plus the cipher overhead, which is `xfrm->minlen`.